src/ert/_c_wrappers/enkf/enkf_main.py

```diff
@@ -88,7 +88,7 @@ def format_seed(random_seed: str):
 
 class EnKFMain:
     def __init__(self, config: "ResConfig", read_only: bool = False):
-        self.config_file = config
+        self.res_config = config
```

Comment:
not against the renaming in itself, but is it a necessary change? If i'm not mistaken - we have consumers of this file which we know nothing about. A change like this could potentially break their scripts?

Reply:
oh, you mean people using ert as a library, and `config_file` is a public field of `EnKFMain`, and this is thus a breaking change of the API?

can we do something like keeping the old name around as an alias, and deprecating it with a warning printed whenever used?

If that sounds reasonable, I'll pull it into its own PR

If we actually think that the naming "config_file" is sound, I'll just scrap the whole rename commit : )

Reply:
I have no idea if anyone actually uses this, but i guess running workflows you will get enkf_main in your lap and it would not surprise me if someone reads the config_file variable. Point is - i don't know. 

> If we actually think that the naming "config_file" is sound, I'll just scrap the whole rename commit : )
I'm not against the new name or refactoring in general - but i tend to be a bit skeptic regarding "unnecessary" refactorings / changes to the codebase. But that is just my opinion 😄

Reply:
okay, let's say this one is "necessary" as it improves readability of the code (which is argueable), how would we go about making this a safe change for the users? does my suggestion above work, or is it too involved? do we ask some users?

Reply:
I think i made myself misunderstood with my comment -> "That i would say is unnecessary. Either we just rename or we don't rename". What I mean is renaming of variables in code should ideally be just a refactoring if we find it to be easier to read or something like that. If we actually need to start thinking of deprecation warnings when renaming a variable we should maybe think twice about if it is actually necessary. I mean - potentially forcing many users to change their code because we have broken an API just because it is better to read the variable-name in some other way is not something I generally support.

Reply:
If we decide to merge this PR though, i would argue the commits should be separated. Actually fixing the issue in this PR as one commit and then refactoring of variable-name in another 👍

Reply:
okay, gotcha, dropping the rename then!